script.skin.helper.skinbackup/default.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+'''
+    script.skin.helper.skinbackup
+    Kodi addon to backup skin settings
+'''
+
+import sys
+from resources.lib.backuprestore import BackupRestore
+from resources.lib.colorthemes import ColorThemes
+from resources.lib.utils import log_exception, log_msg
+
+
+class Main():
+    '''Main entry point for script'''
+
+    def __init__(self):
+        '''Initialization and main code run'''
+        try:
+            self.params = self.get_params()
+            log_msg("called with parameters: %s" % self.params)
+            action = self.params.get("action", "")
+            if not action:
+                # launch main backuprestore dialog
+                BackupRestore().backuprestore()
+            else:
+                # launch module for action provided by this script
+                if hasattr(self, action):
+                    getattr(self, action)()
+                else:
+                    log_msg("No such action: %s" % action, xbmc.LOGWARNING)
+        except Exception as exc:
+            log_exception(__name__, exc)
+        finally:
+            xbmc.executebuiltin("dialog.Close(busydialog)")
+
+    def backup(self):
+        '''backup skin settings to file'''
+        backuprestore = BackupRestore()
+        filters = self.params.get("filter", [])
+        if filters:
+            filters = filters.split("|")
+        silent = self.params.get("silent", "")
+        promptfilename = self.params.get("promptfilename", "") == "true"
+        if silent:
+            silent_backup = True
+            backup_file = silent
+        else:
+            silent_backup = False
+            backup_file = backuprestore.get_backupfilename(promptfilename)
+        backuprestore.backup(filters, backup_file, silent_backup)
+        del backuprestore
+
+    def restore(self):
+        '''restore skin settings from file'''
+        backuprestore = BackupRestore()
+        silent = self.params.get("SILENT", "")
+        if silent and not xbmcvfs.exists(silent):
+            log_msg(
+                "ERROR while restoring backup ! --> Filename invalid."
+                "Make sure you provide the FULL path, for example special://skin/extras/mybackup.zip",
+                xbmc.LOGERROR)
+            return
+        backuprestore.restore(silent)
+
+    def reset(self):
+        '''reset skin settings'''
+        backuprestore = BackupRestore()
+        filters = self.params.get("filter", [])
+        if filters:
+            filters = filters.split("|")
+        silent = self.params.get("silent", "") == "true"
+        backuprestore.reset(filters, silent)
+        xbmc.Monitor().waitForAbort(2)
+        xbmc.executebuiltin("RunScript(script.skin.helper.service,action=checkskinsettings)")
```

Comment:
Hmm, it's optional and should only be executed IF skinhelper service is even present. It's cenrtainly not a dependency. I will add the check in the code, thanks.